src/api/contactGroups.coffee

```diff
@@ -3,148 +3,110 @@ Q = require 'q'
+    utils.logAndSetResponse this, 'forbidden', "User #{this.authenticated.email} is not an admin, API access to addContactGroup denied.", 'info'
+    return
 
-  // Test if the user is authorised
-  if (authorisation.inGroup('admin', this.authenticated) === false) {
-    logger.info('User ' +this.authenticated.email+ ' is not an admin, API access to addContactGroup denied.')
-    this.body = 'User ' +this.authenticated.email+ ' is not an admin, API access to addContactGroup denied.'
-    this.status = 'forbidden';
-    return;
-  }
+  contactGroupData = @request.body
 
-  var contactGroupData = this.request.body
-
-  try {
-    var contactGroup = new ContactGroup(contactGroupData);
-    var result = yield Q.ninvoke(contactGroup, 'save');
+  try
+    contactGroup = new ContactGroup(contactGroupData)
+    result = yield Q.ninvoke(contactGroup, 'save')
     
-    this.body = 'Contact Group successfully created';
-    this.status = 'created';
-    logger.info('User %s created contact group with id %s', this.authenticated.email, contactGroup.id);
-  } catch(e) {
-    logger.error('Could not add a contact group via the API: ' + e);
-    this.body = e.message;
-    this.status = 'bad request';
-  }
-}`
+    utils.logAndSetResponse this, 'created', "Contact Group successfully created", 'info'
+  catch err
+    utils.logAndSetResponse this, 'bad request', "Could not add a contact group via the API: #{err}", 'error'
+
 
 
 #############################################################
 #     Retrieves the details of a specific contact group     #
 #############################################################
-exports.getContactGroup = `function *findContactGroupById(contactGroupId) {
-
-  // Test if the user is authorised
-  if (authorisation.inGroup('admin', this.authenticated) === false) {
-    logger.info('User ' +this.authenticated.email+ ' is not an admin, API access to findContactGroupById denied.')
-    this.body = 'User ' +this.authenticated.email+ ' is not an admin, API access to findContactGroupById denied.'
-    this.status = 'forbidden';
-    return;
-  }
-
-  var contactGroupId = unescape(contactGroupId);
-
-  try {
-    var result = yield ContactGroup.findById(contactGroupId).exec();
-    if (result === null) {
-      this.body = "Contact Group with id '"+contactGroupId+"' could not be found.";
-      this.status = 'not found';
-    } else {
-      this.body = result;
-    }
-  } catch(e) {
-    logger.error('Could not find Contact Group by id '+contactGroupId+' via the API: ' + e);
-    this.body = e.message;
-    this.status = 'internal server error';
-
-  }
-}`
+exports.getContactGroup = (contactGroupId) ->
+  # Must be admin
+  if authorisation.inGroup('admin', @authenticated) is false
+    utils.logAndSetResponse this, 'forbidden', "User #{this.authenticated.email} is not an admin, API access to getContactGroup denied.", 'info'
+    return
+
+  contactGroupId = unescape(contactGroupId)
```

Comment:
here and in a few other places: you can remove the (): `unescape contactGroupId`